tests/functional/compile/test_compile.py

```diff
@@ -0,0 +1,131 @@
+            run_dbt(["compile", "--no-introspect"])
+
+
+class TestEphemeralModels:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "first_ephemeral_model.sql": first_ephemeral_model_sql,
+            "second_ephemeral_model.sql": second_ephemeral_model_sql,
+            "third_ephemeral_model.sql": third_ephemeral_model_sql,
+        }
+
+    def test_first_selector(self, project):
+        run_dbt(["compile", "--select", "first_ephemeral_model"])
+        assert file_exists("first_ephemeral_model")
+        assert not file_exists("second_ephemeral_model")
+        assert not file_exists("third_ephemeral_model")
+
+    def test_middle_selector(self, project):
+        run_dbt(["compile", "--select", "second_ephemeral_model"])
+        assert file_exists("first_ephemeral_model")
+        assert file_exists("second_ephemeral_model")
+        assert not file_exists("third_ephemeral_model")
+
+    def test_last_selector(self, project):
+        run_dbt(["compile", "--select", "third_ephemeral_model"])
+        assert file_exists("first_ephemeral_model")
+        assert file_exists("second_ephemeral_model")
+        assert file_exists("third_ephemeral_model")
+
+    def test_no_selector(self, project):
+        run_dbt(["compile"])
+
+        assert get_lines("first_ephemeral_model") == ["select 1 as fun"]
+        assert get_lines("second_ephemeral_model") == [
+            "with __dbt__cte__first_ephemeral_model as (",
+            "select 1 as fun",
+            ")select * from __dbt__cte__first_ephemeral_model",
+        ]
+        assert get_lines("third_ephemeral_model") == [
+            "with __dbt__cte__first_ephemeral_model as (",
+            "select 1 as fun",
+            "),  __dbt__cte__second_ephemeral_model as (",
+            "select * from __dbt__cte__first_ephemeral_model",
+            ")select * from __dbt__cte__second_ephemeral_model",
+            "union all",
+            "select 2 as fun",
+        ]
+
+
+class TestCompile:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "first_model.sql": first_model_sql,
+            "second_model.sql": second_model_sql,
+            "schema.yml": schema_yml,
+        }
+
+    def test_none(self, project):
+        (results, log_output) = run_dbt_and_capture(["compile"])
+        assert len(results) == 4
+        assert "Compiled node" not in log_output
+
+    def test_inline_pass(self, project):
+        (results, log_output) = run_dbt_and_capture(
+            ["compile", "--inline", "select * from {{ ref('first_model') }}"]
+        )
+        assert len(results) == 1
+        assert "Compiled node 'inline_query' is:" in log_output
+
+    def test_select_pass(self, project):
+        (results, log_output) = run_dbt_and_capture(["compile", "--select", "second_model"])
+        assert len(results) == 3
+        assert "Compiled node 'second_model' is:" in log_output
```

Comment:
I thought in this case we would't print the compiled result for second model?

Reply:
Then `--indirect-selection empty ` is only to make sure we don't compile test etc? makes sense to me!